c2cgeoform/schema.py

```diff
@@ -14,13 +15,30 @@ def deferred_dbsession(node, kw):
     return kw.get('dbsession')
 
 
+def unique_validator(mapper, column, id_column, node, value):
+    dbsession = node.bindings['dbsession']
+    _id = node.bindings['request'].matchdict['id']
+    _id = _id if _id != 'new' else None
+    other_consumer = dbsession.query(mapper).filter(column == value, id_column != _id).first()
+    if other_consumer is not None:
+        exc = colander.Invalid(node, '%s use %s.' % (getattr(other_consumer, id_column.name), value), value)
+        raise exc
+
+
 class GeoFormSchemaNode(SQLAlchemySchemaNode):
 
     def __init__(self, *args, **kw):
         super().__init__(*args, **kw)
         self.request = deferred_request
         self.dbsession = deferred_dbsession
 
+    def add_unique_validator(self, column, column_id):
+        validator = partial(unique_validator, self.class_, column, column_id)
+        if self[column.name].validator is None:
+            self[column.name].validator = validator
+        else:
+            self[column.name].validator = colander.All(self[column.name].validator, validator)
```

Comment:
We do not need to send a request to the database if other validator fails.

Reply:
Sad, it seems that we have to choose between colander.ALL or colander.ANY, and both of them trigger all validators, expecting all to success, or one not to fail, but even if the first has failed (or succeded for ANY), all validators are triggered.